src/protocols/secure_channel/PASESession.h

```diff
@@ -198,7 +201,11 @@ class DLL_EXPORT PASESession : public Messaging::UnsolicitedMessageHandler,
     // mNextExpectedMsg is set when we are expecting a message.
     Optional<Protocols::SecureChannel::MsgType> mNextExpectedMsg;
 
+#if CHIP_CRYPTO_PSA_SPAKE2P
+    Crypto::PSASpake2p_P256_SHA256_HKDF_HMAC mSpake2p;
```

Comment:
Shouldn't CryptoPAL handle this transparently, with `Crypto::Spake2p_P256_SHA256_HKDF_HMAC` being whatever type is needed for the current configuration?

Reply:
We can handle it this way, but if we want to do it properly, we need to modify and split the Spake2p API into a separate file and then we can switch between its implementation. 

Currently, the Spake2p interface assumes that we will use raw-key implementation, whereas in PSA we don't use raw keys at all, so most of the API shouldn't be purely virtual.  The problem is, if I modify the Spake2p API and split it into separate files now the PR will be huge and hard to review. On the other hand, I can provide some #ifdefs in CryptoPAL files to switch between those backends but it will make the code less understandable. 

I guess, for now, we can switch it here and in the next step separate all spake2p backends into separate classes that inherit from the spake2p class.

What do you think about those approaches?

Reply:
I think that's probably fine, as long as there is a concrete plan to remove the ifdef from here.

Reply:
I created a feature request here: https://github.com/project-chip/connectedhomeip/issues/32985. I think we can consider extracting this spake2p API and in this way, we will make the code clearer and allow using raw key and PSA key backends.  After that, we will be able to remove ifdef from here and switch between backends in CryptoPAL.